File: examples/integrations/node/lookup.mjs
const baseUrl = process.env.OHG_BASE_URL ?? "http://127.0.0.1:8787";
const token = process.env.OHG_API_TOKEN;
const identifierType = process.env.OHG_IDENTIFIER_TYPE ?? "mc";
const identifierValue = process.env.OHG_IDENTIFIER_VALUE ?? "123456";

const headers = {
  "Content-Type": "application/json",
};

if (token) {
  headers.Authorization = `Bearer ${token}`;
}

async function request(path, options = {}) {
  const response = await fetch(`${baseUrl}${path}`, {
    ...options,
    headers: {
      ...headers,
      ...options.headers,
    },
  });
  const body = await response.text();

  if (!response.ok) {
    throw new Error(`${response.status} ${response.statusText}: ${body}`);
  }

  return body ? JSON.parse(body) : null;
}

const health = await request("/health", { headers: {} });
console.log("health", health);

const lookup = await request("/v1/carrier/lookup", {
  method: "POST",
  body: JSON.stringify({
    identifier_type: identifierType,
    identifier_value: identifierValue,
    max_age: "24h",
  }),
});

console.log("lookup", {
  report_type: lookup.report_type,
  input: lookup.lookup,
  carrier: {
    usdot_number: lookup.carrier?.usdot_number,
    legal_name: lookup.carrier?.legal_name,
  },
  risk_assessment: lookup.risk_assessment,
});

File: examples/integrations/python/lookup.py
#!/usr/bin/env python3
import json
import os
import urllib.error
import urllib.request


BASE_URL = os.environ.get("OHG_BASE_URL", "http://127.0.0.1:8787")
TOKEN = os.environ.get("OHG_API_TOKEN")
IDENTIFIER_TYPE = os.environ.get("OHG_IDENTIFIER_TYPE", "mc")
IDENTIFIER_VALUE = os.environ.get("OHG_IDENTIFIER_VALUE", "123456")


def request(path, method="GET", payload=None):
    headers = {"Accept": "application/json"}
    body = None

    if payload is not None:
        body = json.dumps(payload).encode("utf-8")
        headers["Content-Type"] = "application/json"

    if TOKEN:
        headers["Authorization"] = f"Bearer {TOKEN}"

    req = urllib.request.Request(
        f"{BASE_URL}{path}",
        data=body,
        headers=headers,
        method=method,
    )

    try:
        with urllib.request.urlopen(req, timeout=30) as response:
            return json.loads(response.read().decode("utf-8"))
    except urllib.error.HTTPError as exc:
        error_body = exc.read().decode("utf-8")
        raise RuntimeError(f"{exc.code} {exc.reason}: {error_body}") from exc


health = request("/health")
print("health", json.dumps(health, indent=2))

lookup = request(
    "/v1/carrier/lookup",
    method="POST",
    payload={
        "identifier_type": IDENTIFIER_TYPE,
        "identifier_value": IDENTIFIER_VALUE,
        "max_age": "24h",
    },
)

print(
    "lookup",
    json.dumps(
        {
            "report_type": lookup.get("report_type"),
            "input": lookup.get("lookup"),
            "carrier": {
                "usdot_number": lookup.get("carrier", {}).get("usdot_number"),
                "legal_name": lookup.get("carrier", {}).get("legal_name"),
            },
            "risk_assessment": lookup.get("risk_assessment"),
        },
        indent=2,
    ),
)
